### PR TITLE
Error manager

### LINKED ISSUE #184: Error Manager
The error manager (previously known as "System Manager") is a module that provides an API to the rest of the system for recording/reporting errors (any function that returns a non-successful error code). This module will send a single error report to the File Transfer Service for downlinking, as well as keep track of error occurances/frequency and perform system resets as necessary (e.g. resetting the OBC if issues with the HAL, resetting the Transceiver if having issues with the Transceiver, etc.)

### PR DESCRIPTION
**Pull request by TheDopplerEffects**
Error manager created but not used by other Modules still

### CODE REVIEW

File: radsat-sk/operation/message/RMessage.c [hunk 1]
```diff
@@ -5,6 +5,7 @@
  */
```
Comment (ATGr8) [inline radsat-sk/operation/message/RMessage.c:5]: append your name

File: radsat-sk/operation/utility/RErrorManager.c [hunk 2]
```diff
@@ -0,0 +1,240 @@
+/**
+ * @file RErrorManager.c
+ * @date April 30, 2022
+ * @author Tyrel Kostyk (tck290)
+ */
+
+#include <RErrorManager.h>
+#include <RFileTransferService.h>
+#include <RTransceiver.h>
+
```
Comment (ATGr8) [inline radsat-sk/operation/utility/RErrorManager.c:10]: remove space

File: radsat-sk/operation/message/RMessage.c (hunk 1 above)
Comment (TheDopplerEffects) [reply radsat-sk/operation/message/RMessage.c:5]: I didn't add anything. Ty said it just needed a PR then I will be adding the error manager functionality to the rest of the code in another PR.

Comment (TheDopplerEffects) [reply radsat-sk/operation/message/RMessage.c:5]: I will add my name if you really think I need to, but I mean to only add my name if I add code.

**Review by IsaacPoirier — APPROVED** [review]
It looks very beautiful, that sure is a lot of changes though holy moly

**Review by ATGr8 — APPROVED** [review]
looks good!

**Pull request merged**